schemas/UnauthenticatedError.json

```diff
@@ -0,0 +1,27 @@
+{
+  "type": "object",
+  "description": "Triggered when the user becomes unauthenticated without actually clicking sign out(E.g. Due to expiry of an access token without a way to refresh).",
+  "properties": {
+    "eventName": {
+      "enum": ["UnauthenticatedError"]
+    },
+    "refreshTokenAuth": {
+      "description": "Whether the auth mechanism is refresh-token-based.",
+      "type": "boolean"
+    },
+    "softLogout": {
+      "description": "Whether a soft logout or hard logout was triggered.",
+      "type": "boolean"
+    },
+    "errorCode": {
+      "description": "The error code as defined in matrix spec.",
+      "type": "string"
+    },
+    "errorReason": {
+      "description": "The reason for the error.",
```

Comment:
what does this contain that's additional to the other properties of the event?

Reply:
The source of this is from the homeserver. If refresh auth for example, it could be a message that explains the the refresh token provided has already been redeemed(they rotate) and is out of date. Or maybe that the format of the refresh token is incorrect. etc,

Reply:
great, can you add that context in the description please?

Reply:
Can't provide an enum for this as the reason can change. but will update the  description to provide that context.